flexboxAPIspecV2.md

```diff
@@ -0,0 +1,783 @@
+# API Details
+
+```c# (but really MIDL3)
+
+runtimeclass FlexboxLayout;
+
+    enum FlexboxDirection
+        {
+            Row = 0,
+            RowReverse = 1,
+            Column = 2,
+            ColumnReverse = 3
+        }
+```
+
+## Flex Wrap
+
+The flex-wrap property specifies whether the flexible items should wrap or not.
+
+Note: If the elements are not flexible items, the flex-wrap property has no effect.
+
+```xml
+
+<controls:LayoutPanel MaxWidth="500">
+    <controls:LayoutPanel.Layout>
+        <controls:FlexboxLayout Direction="Row" Wrap="Wrap"/>
+    </controls:LayoutPanel.Layout>
+            <Grid MinWidth="10">
+                <TextBlock>1</TextBlock>
+            </Grid>
+            <Grid MinWidth="20">
+                <TextBlock>2</TextBlock>
+            </Grid>
+            <Grid MinWidth="100">
+                <TextBlock>3</TextBlock>
+            </Grid>
+            <Grid MinWidth="120">
+                <TextBlock>4</TextBlock>
+            </Grid>
+            <Grid MinWidth="120">
+                <TextBlock>5</TextBlock>
+            </Grid>
+            <Grid MinWidth="60">
+                <TextBlock>6</TextBlock>
+            </Grid>
+            <Grid MinWidth="60">
+                <TextBlock>7</TextBlock>
+            </Grid>
+            <Grid MinWidth="100">
+                <TextBlock>8</TextBlock>
+            </Grid>
+        </controls:LayoutPanel>
+```
+### Output of the above code (styling omitted)
+![image](./FlexWrap.PNG)
+
+
+
+##  Flex Wrap Accepted Values
+
+| Value | Description |
+|-|-|
+| NoWrap (Default) | Default value. Specifies that the flexible items will not wrap |
+| Wrap | Specifies that the flexible items will wrap if necessary	|
+| WrapReverse | Specifies that the flexible items will wrap, if necessary, in reverse orde |
+
+
+# API Details
+
+```c# (but really MIDL3)
+
+runtimeclass FlexboxLayout;
+
+    enum FlexboxWrap
+        {
+            NoWrap = 0,
+            Wrap = 1,
+            WrapReverse = 2,
+        }
+```
+
+
+## Flex JustifyContent
+
+The justify-content property aligns the flexible container's items when the items do not use all available space on the main-axis (horizontally).
+
+```xml
+
+<controls:LayoutPanel>
+    <controls:LayoutPanel.Layout>
+        <controls:FlexboxLayout JustifyContent="Start"/>
+    </controls:LayoutPanel.Layout>
+            <Grid MinWidth="120"/>
+            <Grid MinWidth="60"/>
+            <Grid MinWidth="120"/>
+</controls:LayoutPanel>
+```
+
+### Output of the above code (styling omitted)
+![image](./FlexJustifyContentStart.PNG)
+
+##  Flex JustifyContent Accepted Values
+
+| Value | Description |
+|-|-|
+| Start (Default) | Items are positioned at the beginning of the container |
+| End | Items are positioned at the end of the container|
+| Center | Items are positioned in the center of the container |
+| SpaceBetween | Items will have space between them |
+| SpaceAround | Items are positioned at the end of the container|
```

Comment:
This description doesn't match the name to me, is it written correctly? Might need more examples.